src/Shared/IdentityClient/Extensions/UserExtensions.cs

```diff
@@ -1,20 +1,41 @@
-﻿using System.Linq;
-
-using Travely.IdentityClient.Authorization.Data;
-
-namespace Microsoft.AspNetCore.Http
-{
-    public static class UserExtensions
-    {
-        public static UserInfo GetTravelyUserInfo(this HttpContext httpContext)
-        {
-            UserInfo userInfo = new UserInfo();
-            foreach(var prop in userInfo.GetType().GetProperties())
-            {
-                prop.SetValue(userInfo, httpContext.User.Claims.FirstOrDefault(claim => claim.Type == prop.Name)?.Value);
-            }
-
-            return userInfo;
-        }
-    }
-}
+﻿using System;
+using System.ComponentModel;
+using System.Linq;
+
+using Travely.IdentityClient.Authorization.Data;
+
+namespace Microsoft.AspNetCore.Http
+{
+    public static class UserExtensions
+    {
+        public static UserInfo GetTravelyUserInfo(this HttpContext httpContext)
+        {
+            UserInfo userInfo = new UserInfo();
+            foreach(var propInfo in userInfo.GetType().GetProperties())

```

Comment:
This is very bad code. Why are you using reflection? Reflection usage is not justified here.

Do simple assignment. Create following 
```
public static class TravelyClaims {
   public const string Role = "role";
   public const string UserId = "sub";
   public const string AgencyId= "agency_id";
// e.t.c
// for exact claim values sync with @erick20 
}
```
and create UserInfo object in following simple manner
```
var userInfo = new UserInfo {
   UserId = Claims.FirstOrDefault(claim => claim.Type == TravelyClaims.UserId)?.Value,
   AgencyId= Claims.FirstOrDefault(claim => claim.Type == TravelyClaims.AgencyId)?.Value,
// e.t.c
}

```